src/main/scala/is/hail/utils/IntervalTree.scala

```diff
@@ -18,9 +18,18 @@ case class Interval(start: Any, end: Any, includeStart: Boolean, includeEnd: Boo
     (compareStart > 0 || (includeStart && compareStart == 0)) && (compareEnd < 0 || (includeEnd && compareEnd == 0))
   }
 
+  //like contains, but excluding endpoints
+  private def containsWithin(pord: ExtendedOrdering, position: Any): Boolean = {
+    val compareStart = pord.compare(position, start)
+    val compareEnd = pord.compare(position, end)
+    compareStart > 0 && compareEnd < 0
+  }
+
   def overlaps(pord: ExtendedOrdering, other: Interval): Boolean = {
```

Comment:
So overlaps is another tricky one like isEmpty that isn't precise, since: [a, b+1), (b, c] doesn't overlap but b+1 > b.  At some point we should probably change the name of overlaps and isEmpty to indicate they can return a false negative.